### PR TITLE
Update documentation for testing CodeQL queries in VS Code

### PR DESCRIPTION
**Pull request by koesie10**
This updates the documentation for testing CodeQL queries in VS Code using the CodeQL extension to use the native test UI of VS Code rather than the third-party Test Explorer extension. Support for the Test Explorer extension has been removed from the extension since v1.12.1 (31 January 2024).

/cc @subatoi

### CODE REVIEW

**Review by shati-patel — COMMENTED** [review]
Nice! Thanks for fixing this ❤️

Quick question: should we call this the "Testing" view instead of "Test Explorer"? (I don't think "Test Explorer" appears in the VS Code native UI now)

File: docs/codeql/codeql-for-visual-studio-code/testing-codeql-queries-in-visual-studio-code.rst [hunk 1]
```diff
@@ -12,27 +12,31 @@ About testing queries in VS Code
...
 
-1. Open the Test Explorer view in the sidebar.
+1. Open the **Test Explorer** view in the sidebar.
 
    .. image:: ../images/codeql-for-visual-studio-code/open-test-explorer.png
       :width: 350
       :alt: Open the Test Explorer view
 
 2. To run a specific test, hover over the file or folder name and click the play button. To run all tests in your workspace, click the play button at the top of the view. If a test takes too long to run, you can click the stop button at the top of the view to cancel the test.
-3. The icons show whether a test passed or failed. If it failed, right-click the failed test and click **CodeQL: Show Test Output Differences** to display the differences between the expected output and the actual output.
```
Comment (shati-patel) [inline docs/codeql/codeql-for-visual-studio-code/testing-codeql-queries-in-visual-studio-code.rst:29]: Side note: I see that we still have the "Show Test Output Differences" command in the [extension code](https://github.com/github/vscode-codeql/blob/2ba23ceead55208e616587be12696bcc9a28be66/extensions/ql-vscode/src/query-testing/test-manager.ts#L217), but it doesn't seem to be used anywhere 😅 I wonder if we can delete it 👀 🤔

**Comment by koesie10** [comment]
> Quick question: should we call this the "Testing" view instead of "Test Explorer"? (I don't think "Test Explorer" appears in the VS Code native UI now)

I'm not sure. I based the naming on [the Python testing documentation](https://code.visualstudio.com/docs/python/testing#_configure-tests) from VS Code itself, but it seems like the view is called "Testing" within VS Code. However, all documentation for VS Code itself uses Test Explorer: [Java (Testing Explorer)](https://code.visualstudio.com/docs/java/java-testing#_enable-testing-and-adding-test-framework-jars-to-your-project), [C#](https://code.visualstudio.com/docs/csharp/testing#_test-explorer), and [Testing API](https://code.visualstudio.com/api/extension-guides/testing).

File: docs/codeql/codeql-for-visual-studio-code/testing-codeql-queries-in-visual-studio-code.rst (hunk 1 above)
Comment (koesie10) [reply docs/codeql/codeql-for-visual-studio-code/testing-codeql-queries-in-visual-studio-code.rst:29]: I'm not sure if this code is reachable, but it seems like we should remove it if it's not.

**Comment by subatoi** [comment]
> I'm not sure. I based the naming on [the Python testing documentation](https://code.visualstudio.com/docs/python/testing#_configure-tests) from VS Code itself, but it seems like the view is called "Testing" within VS Code. However, all documentation for VS Code itself uses Test Explorer: [Java (Testing Explorer)](https://code.visualstudio.com/docs/java/java-testing#_enable-testing-and-adding-test-framework-jars-to-your-project), [C#](https://code.visualstudio.com/docs/csharp/testing#_test-explorer), and [Testing API](https://code.visualstudio.com/api/extension-guides/testing).

I noticed the same discrepancy, but I think the documentation for VS Code itself is in fact wrong :(

**Review by shati-patel — APPROVED** [review]
Looks good to me! Thanks for picking this up so quickly ⚡ 🙇🏽‍♀️

**Review by subatoi — APPROVED** [review]
(Just in case you wanted my review)—thank you very much!

**Pull request merged**